pkgs/applications/audio/exaile/default.nix

```diff
@@ -0,0 +1,110 @@
+{ stdenv, lib, fetchFromGitHub
+, gobject-introspection, makeWrapper, wrapGAppsHook
+, gtk3, gst_all_1, python3
+, gettext ? null, gnome3 ? null, help2man ? null, keybinder3 ? null, libnotify ? null, streamripper ? null, udisks ? null, webkitgtk ? null
+, iconTheme ? gnome3.adwaita-icon-theme
+, deviceDetectionSupport ? true
+, documentationSupport ? true
+, notificationSupport ? true
+, scalableIconSupport ? true
+, translationSupport ? true
+, bpmCounterSupport ? false
+, ipythonSupport ? false
+, lastfmSupport ? false
+, lyricsManiaSupport ? false
+, lyricsWikiSupport ? false
+, multimediaKeySupport ? false
+, musicBrainzSupport ? false
+, podcastSupport ? false
+, streamripperSupport ? false
+, wikipediaSupport ? false
```

Comment:
Should we enable them by default?

Reply:
Maybe? The options I defaulted to false are all plugins that are disabled by default and can be enabled in the gui so the only real consequence is closure size. If they're all going to be true by default perhaps they should be renamed `disableX ? false` per [this discussion](https://discourse.nixos.org/t/naming-optional-features-and-dependencies-in-nixpkgs-enablefoo-usefoo-withfoo-supportfoo-foosupport-or/9753/3).

Reply:
@ryneeverett Adding too many options is not very productive as we are not build testing all of the combinations. A power user can always do something like `exaile.override { webkitgtk = null; }`.

Reply:
@veprbl I guess I don't really accept the premise that it's a package manager's job to test whether the optional dependencies -- as declared by upstream -- are truly optional. Also isn't your argument equally valid against *any* number of options?

Reply:
Also, I ran some builds to compare closure size:

- All options false -> 583mb
- Current default options -> 832mb
- All options true -> 1.5gb

Nearly doubling the closure size in order to install the dependencies of all the plugins (which are disabled by default) seems crazy. Especially when there's no reason to believe upstream is done adding plugins.

Also, particularly for the optional python dependencies, it seems improbable that they have any effect at build time.

Reply:
> I guess I don't really accept the premise that it's a package manager's job to test whether the optional dependencies -- as declared by upstream -- are truly optional.

When you write an API with a bunch of knobs you usually want those to be tested. In the current setup there is not even a test that exile's build system accepts the libraries that we may pass to it. It will be up to a user to discover if something is broken.

> Also isn't your argument equally valid against any number of options?

Not sure what you mean by that.

Reply:
> In the current setup there is not even a test that exile's build system accepts the libraries that we may pass to it.

Is there a simple way to test packages with different combinations of inputs? I'm not familiar, though a hack would be to add another top level package `exaileWithPlugins` in which all the flags are true.

>> Also isn't your argument equally valid against any number of options?

> Not sure what you mean by that.

I'm working from the assumption that options are not generally tested and if that's the case then it's not clear that many options are worse in principle than fewer options.

Reply:
> > In the current setup there is not even a test that exile's build system accepts the libraries that we may pass to it.
> 
> Is there a simple way to test packages with different combinations of inputs? I'm not familiar, though a hack would be to add another top level package `exaileWithPlugins` in which all the flags are true.

Such package is often called `exaileFull` (e.g. `gitFull`).

> > > Also isn't your argument equally valid against any number of options?
> 
> > Not sure what you mean by that.
> 
> I'm working from the assumption that options are not generally tested and if that's the case then it's not clear that many options are worse in principle than fewer options.

More options means more combinations to test/debug, or, if you like, combinations left untested.